draft-packages/form/KaizenDraft/Form/Primitives/FieldMessage/FieldMessage.tsx

```diff
@@ -36,24 +37,23 @@ const FieldMessage: FieldMessage = ({
   reversed = false,
   position = "bottom",
 }) => {
-  const textColor = reversed
-    ? status === "error"
-      ? "dark-reduced-opacity"
-      : "white-reduced-opacity"
-    : "dark-reduced-opacity"
+  const textColor =
+    status === "default"
+      ? reversed
+        ? "white-reduced-opacity"
+        : "dark-reduced-opacity"
+      : "dark"
   return (
     <div
       id={id}
       data-automation-id={automationId}
-      className={classnames(styles.message, {
+      className={classnames(styles.message, styles[status], {
         [styles.reversed]: reversed,
-        [styles.default]: status === "default",
-        [styles.error]: status === "error",
         [styles.positionBottom]: position === "bottom",
         [styles.positionTop]: position === "top",
       })}
     >
-      {status === "error" && warningIcon}
+      {(status === "error" || status === "caution") && warningIcon(reversed)}
```

Comment:
Would it be more React-idiomatic to use a component here rather than a function?
E.g. `... && <WarningIcon reversed={reversed} />`
It would just be a simple functional component private to this file.

Reply:
Also, I'd love to see a unit test here for when it shows the icon and when not.
There might be a future where we test this with Chromatic, but I think for now we should have at least some rudimentary Jest unit tests.

Reply:
Sounds good, I was tossing up whether or not it warranted a test, but will add that in